### PR TITLE
"hey what if we could flay people from a distance with a gun"

### PR DESCRIPTION
**Pull request by lolman360**
## About The Pull Request

Adds the Man-Portable Flensing Apparatus, for flaying people at a distance and in style. 12 tc in the uplink. 

Fires razor nets at people, which work in a similar way to clockwork skewers (i.e copypasta code.)

Also turns people into husks if they have over 300 bruteloss. Does about 50 brute, has one shot. Can only "recharge" by retracting nearby nets (7 tile range, in view.)

Also adds the flensing-gun (harvesting) which generates "synth"flesh on flense successes. altclick it to splash the internal reagent store on yourself.

needs sprites. has placeholder ones atm

## Why It's Good For The Game

haha skin go `sound/misc/desceration-03.ogg`

## Changelog
:cl:
add: The Syndicate have recently "found" a shipment of Carminus Yards flensing-guns. Make sure your skin is secure.
/:cl:

### CODE REVIEW

**Comment by lolman360** [comment]
https://cdn.discordapp.com/attachments/161574200581029888/742605516429328384/desceration-03.ogg

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -336,4 +336,51 @@
...
+	can_charge = FALSE
+	ammo_type = list(/obj/item/ammo_casing/energy/flenser)
+	cell_type = /obj/item/stock_parts/cell/emergency_light //120 charge
+
+/obj/item/gun/energy/warcrime/attack_self(mob/user)
+	to_chat(user, "<span class='danger'>You retract all nearby nets using the electromagnetic tether!</span>")
+	for(var/obj/structure/flensingnet/F in oview(7, user))
+		F.rend()
+		addtimer(CALLBACK(F, .proc/take_damage, max_integrity), 1)
+		cell.charge = 120
```
Comment (SmArtKar) [inline code/modules/projectiles/guns/energy/special.dm:360]: cell.charge += 120

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -336,4 +336,51 @@
...
+	return .
+
+/obj/item/gun/energy/warcrime/unethical/attack_self(mob/user)
+	to_chat(user, "<span class='danger'>You retract all nearby nets using the electromagnetic tether!</span>")
+	for(var/obj/structure/flensingnet/F in oview(7, user))
+		F.rend()
+		if(F.rended)
+			reagents.add_reagent(/datum/reagent/medicine/synthflesh/flesh, 10)
+		addtimer(CALLBACK(F, .proc/take_damage, max_integrity), 1)
+		cell.charge = 120
```
Comment (SmArtKar) [inline code/modules/projectiles/guns/energy/special.dm:381]: cell.charge += 120

File: code/modules/projectiles/guns/energy/special.dm [hunk 3]
```diff
@@ -336,4 +336,51 @@
...
+
+/obj/item/gun/energy/warcrime/unethical/AltClick(mob/user)
+	to_chat(user, "<span class='notice'>You splash the contents of the flesh storage onto yourself. Gross.</span>")
```
Comment (SmArtKar) [inline code/modules/projectiles/guns/energy/special.dm:384]: Maybe you should make a visible_message for that?

File: code/modules/projectiles/projectile/special/flense.dm [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		var/turf/Tloc = get_turf(target)
+		if(!locate(/obj/structure/flensingnet) in Tloc)
+			new /obj/structure/flensingnet(Tloc)
+	return ..()
+
+/obj/structure/flensingnet
+	name = "flensing net"
+	desc = "A net made of concertina wire."
+	icon = 'icons/obj/iv_drip.dmi'
+	icon_state = "iv_drip"
```
Comment (SmArtKar) [inline code/modules/projectiles/projectile/special/flense.dm]: You need a better sprite for this one

File: code/modules/projectiles/projectile/special/flense.dm [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+			playsound(whale, 'sound/misc/desceration-03.ogg', 50, TRUE)
+			whale.adjustBruteLoss(10)
+		mouse_opacity = MOUSE_OPACITY_OPAQUE //So players can interact with the tile it's on to pull them off
+		buckle_mob(whale, TRUE)
+
+/obj/structure/flensingnet/process()
+	if(buckled_mobs && LAZYLEN(buckled_mobs))
+		var/mob/living/spitroast = buckled_mobs[1]
+		spitroast.adjustBruteLoss(0.1)
+		if(spitroast.health < spitroast.maxHealth/1.3)//around 153 hp on a 200hp mob
```
Comment (SmArtKar) [inline code/modules/projectiles/projectile/special/flense.dm:82]: Totally balanced. At least remove projectile's damage

**Review by SmArtKar — CHANGES_REQUESTED** [review]
This shit is too unbalanced. Basically it's insta-stun with a lot of brute damage.

File: code/modules/projectiles/projectile/special/flense.dm (hunk 4 above)
Comment (lolman360) [reply code/modules/projectiles/projectile/special/flense.dm]: >needs sprites. has placeholder ones atm

File: code/modules/projectiles/projectile/special/flense.dm (hunk 5 above)
Comment (lolman360) [reply code/modules/projectiles/projectile/special/flense.dm:82]: yes 0.05 brute/sec is very op

**Comment by lolman360** [comment]
i forget how stupid knockdowns are i've reduced the length of the user-procable one to 0.5s

if you resist out or are helped out it's 3s and if the net is destroyed without doing the rend() proc it knockdowns for 10s

it doesn't stun anything apart from borgs which can be removed

**Pull request closed without merging**